joomla-projects/gsoc19_page-builder
# [WIP] Initial setup for the page builder.

### Summary of Changes
This PR introduces the new feature Page Builder. With Apodis, the new frontend template, users are able to define the structure of their website with an editor. It allows to drag and drop the elements grid, column, container, module position, message and component into the order and position the user wants. These elements are plugins and therefore expandable and (de-) activatable.

### Testing Instructions
Log in to the Joomla backend and go to System -> Site Template Styles.
Set Apodis as the default frontend template and look into its settings.
Here you should find the Page Builder tab and the editor.
Use the editor, try it out, add, edit and remove elements. Save and reload. Change the target device. Test module positions like 'sidebar-right' or 'footer' and assign modules (outside of the Page Builder).
Look at the frontend. Does Apodis look like you expected with the elements set in the editor? Have the settings of each element, e.g. classes or gaps, the correct effect? Is it responsive and adjust on different screen sizes? Are there any browser console errors caused by JavaScript?

### Documentation Changes Required
Yes. You can find it in the [Joomla Documentation](https://docs.joomla.org/J4.x:Page_Builder).


This is draft Pull request.

**Don't merge this PR this should be created by students.**
